### ISSUE TITLE
[BUG] CLIENT SPOOF NEXTGEN/LEGACY - Custom and other modes don't work properly

### ISSUE THREAD

**Issue by VilaED** [CCBlueX/LiquidBounce#4086]
### LiquidBounce Branch

Both

### LiquidBounce Build/Version

v0.15.2 NEXTGEN - b98 Legacy 

### Operating System

Windows 10

### Minecraft Version

1.8.9.+

### Describe the bug

Not even a version of liquidbounce with client spoof is working, the branch bypass I'm trying on this rust server minebuilders.me and when I select a mode in client spoof or deactivate anti forge it is not possible to enter the "rust v2 or other game" game mode it gives me It accuses me of using fabric/forge and not being able to enter it, I come to the conclusion that the payload is just alternating the name but not making the necessary changes. please correct

### Steps to reproduce

- go connect to server: minebuilders.me (using this one because I know that it blocks forge/fabric and clients in addition to labymod and their client, however, not even vanilla can enter because liquidbounce does not bypass)
- select rust v2 game mode
- you won't be able to get in
- 
![image](https://github.com/user-attachments/assets/e5266425-560f-4d85-bf64-6ad679b176ac)

### Client Log

```shell
[09:59:02] [Render thread/INFO]: [CHAT] You cannot use this client to play Minebuilders!
[09:59:02] [Render thread/INFO]: [CHAT] There's too many issues with the client you're using that will cause instability in-game. Please use Vanilla, LabyMod, or Minebuilders client.
[09:59:04] [Render thread/INFO]: [CHAT] Please pick a server from the menu!
[09:59:06] [Render thread/INFO]: [CHAT] You cannot use this client to play Minebuilders!
[09:59:06] [Render thread/INFO]: [CHAT] There's too many issues with the client you're using that will cause instability in-game. Please use Vanilla, LabyMod, or Minebuilders client.
[09:59:06] [Render thread/INFO]: [CHAT] Please pick a server from the menu!
[09:59:08] [Render thread/INFO]: [CHAT] Please pick a server from the menu!
[09:59:10] [Render thread/INFO]: [CHAT] Please pick a server from the menu!
[09:59:11] [Render thread/INFO]: [CHAT] You cannot use this client to play Minebuilders!
[09:59:11] [Render thread/INFO]: [CHAT] There's too many issues with the client you're using that will cause instability in-game. Please use Vanilla, LabyMod, or Minebuilders client.
[09:59:12] [Render thread/INFO]: [CHAT] Please pick a server from the menu!
[09:59:14] [Render thread/INFO]: [CHAT] Please pick a server from the menu!
[09:59:16] [Render thread/INFO]: [CHAT] Please pick a server from the menu!
[09:59:18] [Render thread/INFO]: [CHAT] Please pick a server from the menu!
[09:59:20] [Render thread/INFO]: [CHAT] Please pick a server from the menu!
[09:59:22] [Render thread/INFO]: [CHAT] Please pick a server from the menu!
[09:59:24] [Render thread/INFO]: [CHAT] Please pick a server from the menu!
```

### Screenshots

![image](https://github.com/user-attachments/assets/892912c4-aae9-4613-9009-e135ade843b1)

**Comment by VilaED** [comment]
@1zun4 bot added tag for legacy but it is for nextgen too

**Comment by ccetl** [comment]
In nextgen it successfully spoofs the brand initially but there might be packets about the registry sent from fabric. This shouldn't be an issue on most servers tho since most servers just check for the brand.
An easy fix would probably be blocking all packets that contain the word "fabric" or something.

**Comment by VilaED** [comment]
> Contributor

yes it is bad because staffs /plugins and good anticheats can detect it i use flux and flux can bypass but nextgen no 

@1zun4 @mems01 @superblaubeere27 @opZywl @ccetl 

has solution?

**Comment by opZywl** [comment]
Hi, you is slayermaster on discord right?
I remember talking to you on discord about this subject

**Comment by 1zun4** [comment]
![image](https://github.com/user-attachments/assets/603b5acc-5cef-4f69-995b-96646191b4d2)

I'm not even using a VPN... I had to use LiquidProxy Dedicated (VPN) to join their server + pass the AntiBot.

![image](https://github.com/user-attachments/assets/61aa937b-80e2-438e-96a2-e956d4d75c55)

But now I am stuck:
![image](https://github.com/user-attachments/assets/f6c1f1f4-8a2c-4ab0-bc66-ac4a21062d87)

What ever I click, nothing happens. Do you have any better server to try this on? I also feel like this issue you reported only is caused by ViaFabricPlus 1.8.X translation. Please tell me if you use protocol translation.

**Comment by opZywl** [comment]
> 

I played this minigame as well; to access it, select the Rust V2 mode.

However, he forgot to mention that you can join the game using either the Vanilla client or the Minebuilders client. You can download the client to check its branding, but I believe that certain methods, like ViaFabric or protocols used by LiquidBounce, can't bypass the server restrictions—this includes the Legacy protocol.

I can access the game mode through the Legacy protocol. I remember that in the past, you had managed to bypass the restrictions using a resource pack spoof, but it looks like they’ve patched that method now.

And disable antiforge in legacy for join but legacy dont bypass rust v2 .-.

**Comment by opZywl** [comment]
@VilaED i send video for 1zuna trying join using nextgen
![image](https://github.com/user-attachments/assets/ea92054b-4c07-46b3-b679-dbe78cace149)

**Comment by opZywl** [comment]
legacy: https://youtu.be/C1GohvGp6z4 

but v2 mode dont play

**Comment by localpthebest** [comment]
> In nextgen it successfully spoofs the brand initially but there might be packets about the registry sent from fabric. This shouldn't be an issue on most servers tho since most servers just check for the brand. An easy fix would probably be blocking all packets that contain the word "fabric" or something.

ye i know that forge sends a packet with a list of all installed mods to the server maybe its the same for fabric?

**Comment by ccetl** [comment]
> > In nextgen it successfully spoofs the brand initially but there might be packets about the registry sent from fabric. This shouldn't be an issue on most servers tho since most servers just check for the brand. An easy fix would probably be blocking all packets that contain the word "fabric" or something.
> 
> ye i know that forge sends a packet with a list of all installed mods to the server maybe its the same for fabric?

There are (packet IDs) "fabric:custom_ingredient_sync", "c:register", "c:version", "minecraft:register". "minecraft:unregister", "fabric:registry/sync/complete", "fabric:registry/sync/direct", possible more, possible less sent by fabric.
Some mods might send their own packets.
All those packets might expose the client as being fabric to the server.

**Comment by localpthebest** [comment]
> > > In nextgen it successfully spoofs the brand initially but there might be packets about the registry sent from fabric. This shouldn't be an issue on most servers tho since most servers just check for the brand. An easy fix would probably be blocking all packets that contain the word "fabric" or something.
> > 
> > 
> > ye i know that forge sends a packet with a list of all installed mods to the server maybe its the same for fabric?
> 
> There are (packet IDs) "fabric:custom_ingredient_sync", "c:register", "c:version", "minecraft:register". "minecraft:unregister", "fabric:registry/sync/complete", "fabric:registry/sync/direct", possible more, possible less sent by fabric. Some mods might send their own packets. All those packets might expose the client as being fabric to the server.

yeesh thats quite a lot to get rid of

**Comment by EclipsesDev** [comment]
@VilaED in legacy it seems lunarclient brand spoof works just fine

**Comment by ghost** [comment]
Just disable `Block Resource Pack Exploit`.
